src/main.rs

```diff
@@ -137,12 +156,22 @@ impl State {
             .expect("Unable to create surface");
 
         // Create the EGL surface.
-        let config = context.config_id();
-        let native_surface = WlEglSurface::new(surface.id(), self.size.width, self.size.height)
-            .expect("Unable to create EGL surface");
-        let pixel_format = context.pixel_format().expect("No valid pixel format present");
-        let egl_surface = EGLSurface::new(&display, pixel_format, config, native_surface, None)
-            .expect("Unable to bind EGL surface");
+        let mut raw_window_handle = WaylandWindowHandle::empty();
+        raw_window_handle.surface = surface.id().as_ptr().cast();
+        let raw_window_handle = RawWindowHandle::Wayland(raw_window_handle);
+        let surface_attributes = SurfaceAttributesBuilder::<WindowSurface>::new().build(
+            raw_window_handle,
+            NonZeroU32::new(self.size.width as u32).unwrap(),
+            NonZeroU32::new(self.size.height as u32).unwrap(),
+        );
+
+        let egl_surface = unsafe {
+            display
+                .create_window_surface(&config, &surface_attributes)
+                .expect("Failed to create EGL surface")
+        };
+
+        let context = context.make_current(&egl_surface).expect("Failed to make context current");
```

Comment:
Should the context maybe be made current on creation? Not sure but seems like a lot of times you'd probably want that.

Reply:
No, because you can create on one thread and send to other thread. The point is to have sendable context on creation, and making context current flushes the current context, which could be undesired.